plugins/httpapi/aci.py

```diff
@@ -325,3 +324,15 @@ def cert_auth(self, method, path, payload=""):
             + "APIC-Request-Signature={0}".format(to_native(base64.b64encode(sig_signature)))
         )
         return headers
+
+    def create_abspath(self, working_dir, file):
+        """
+        Works if variable 'file' is:
+        - a filename which is in the same working directory as the ansible playbook,
+        - a relative path of the file that is either in the same folder or an adjacent folder as the ansible playbook,
+        - an absolute path of the file.
+        Doesn't work for more complex folder structures.
+        """
+        l = len(file.split("/"))
+        s = ("/").join(".." for i in range(l - 1))
```

Comment:
I am not sure what the above two lines of code are doing.
If the relative path of the file has more than two items as seen here -> private_key: "dir/sub_dir/admin.key"
 os.path.abspath would resolve to an incorrect absolute path with respect to the working directory.


```
def create_abspath(self, working_dir, file):
   return os.path.abspath(os.path.join(working_dir, file))
```

Reply:
It also works for the specific case you mentioned only if it's either in the current working directory or in an adjacent folder like this for example:
`
main_dir:
  sub_dir_1:
    admin.key
  sub_dir_2:
    playbook.yml
`
 The function `os.getcwd()` will get us `/{{all the parent dir}}/main_dir/sub_dir_2`
and the function will return
`python
 os.path.abspath(os.path.join( os.getcwd(), ../.. , main_dir/sub_dir_1/admin.key)
`
which will give us the expected result we want `/{{all the parent dir}}/main_dir/sub_dir_1/admin.key`

There are only three possible cases for the function to work as expected:

1.  If it's a filename or relative path of a file which is in the same dir as the playbook
2.  If it's a relative path of a file that is as a adjacent folder as one the playbook is in. This mean the playbook and the file we want to use must have the same level of path depth.
3. If it's a absolute path already

I tried every case and mostly in case 2, I tried different length of relative path and it works correctly. If we want to handle more complex folder structure like non-adjacent folders with relative path or just filename, we need to use the backward find path function I was using in a previous commit.